go-runtime/compile/schema.go

```diff
@@ -436,6 +575,39 @@ func visitStruct(pctx *parseContext, node ast.Node, tnode types.Type) (*schema.D
 	return dataRef, nil
 }
 
+func visitValue(enum *schema.Enum, node ast.Node, tnode types.Type) (schema.Value, error) {
+	switch t := node.(type) {
+	case *ast.BasicLit:
+		switch underlying := tnode.Underlying().(type) {
+		case *types.Basic:
+			switch underlying.Kind() {
+			case types.String:
+				value, err := strconv.Unquote(t.Value)
+				if err != nil {
+					return nil, err
+				}
+				return &schema.StringValue{Pos: goPosToSchemaPos(node.Pos()), Value: value}, nil
+
+			case types.Int, types.Int64:
+				value, err := strconv.ParseInt(t.Value, 10, 64)
+				if err != nil {
+					return nil, err
+				}
+				return &schema.IntValue{Pos: goPosToSchemaPos(node.Pos()), Value: int(value)}, nil
+			default:
+				return nil, fmt.Errorf("%s: unsupported basic type %s", goPosToSchemaPos(node.Pos()),
+					underlying)
+			}
+		default:
+			return nil, fmt.Errorf("%s: unsupported type %T", goPosToSchemaPos(node.Pos()), node)
+		}
+	case *ast.Ident:
```

Comment:
What's this case for? Could you please add a few comments in these visitor functions to illustrate what Go syntactical elements these are matching?

Reply:
simplified this logic, should be more straightforward now